### PR TITLE
Making py3 compatible

### PR DESCRIPTION
**Pull request by bsipocz**
~~There are still 132 tests failing locally.~~ Finished this off on the plane, and now rebased to get rid of the conflicts.

The changes include some automated ones using 2to3. Also manual ones, including some pep8 issues that were causing errors (e.g. E101, the mixture of tab and space indents), etc, and the most serious ones being the int float divisions. I think I managed to use the appropriate one everywhere.

Also sorry for the white space changes, my editor does that automatically.

### CODE REVIEW

**Comment by bsipocz** [comment]
Finished this off on the plane, and now rebased to get rid of the conflicts.
Locally all test pass, on travis still there are some weird things going on. Most probably I can iron this out over the weekend.

**Comment by aphearin** [comment]
Wow @bsipocz - this is an outstanding update! Many thanks all around, but *especially* for the nitty gritty work involved here, such as fixing the float divisions. Looking through the int/float commit, all changes do look correct by eye but there are a few changes that I'm not confident would be caught by the test suite. I just caught a minor one in double_tree; I'll PR against yours within the next day or two.

File: halotools/mock_observables/pair_counters/double_tree.py [hunk 1]
```diff
@@ -440,76 +443,76 @@ def set_tree2_cell_sizes(self,
         y2mult = min(max_cells_per_dimension, y2mult)
         z2mult = min(max_cells_per_dimension, z2mult)
 
-        modified_x2_cellsize = self.tree1.xcell_size/x2mult
-        modified_y2_cellsize = self.tree1.ycell_size/y2mult
-        modified_z2_cellsize = self.tree1.zcell_size/z2mult
+        modified_x2_cellsize = self.tree1.xcell_size / x2mult
```
Comment (bsipocz) [inline halotools/mock_observables/pair_counters/double_tree.py]: @aphearin  - you mean this one not being int division? If yes then this was intentional as I though this was a bug in the original code (based to the docstring these cellsizes should be floats, and indeed similar parameters in other functions were with float division). So I'm sorry if I got it wrong.

Comment (aphearin) [inline halotools/mock_observables/pair_counters/double_tree.py]: No, no @bsipocz - your fix was correct. Cell sizes are indeed floats. The minor pr I submitted earlier (which was probably overkill) was all I meant.

**Comment by aphearin** [comment]
I'm going to close this since I've updated it #462

**Pull request closed without merging**